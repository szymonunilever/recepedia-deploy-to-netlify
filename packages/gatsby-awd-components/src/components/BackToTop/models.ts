import { ReactNode } from 'react';
import { ButtonViewType } from '../Button';
import { UnileverLibraryComponent } from '../../models';

export interface BackToTopProps
  extends UnileverLibraryComponent<AppContent.BackToTopContent> {
  Icon?: JSX.Element;
  children?: ReactNode | ReactNode[];
  viewType?: ButtonViewType;
  hideTopPositionPx?: number;
}

export interface CustomEventTarget {
  scrollingElement: HTMLElement | null;
}
