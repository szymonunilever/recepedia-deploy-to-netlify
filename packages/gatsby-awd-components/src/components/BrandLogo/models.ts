export interface BrandLogoProps {
  className?: string;
  brand?: string;
  linkTo?: string;
}

