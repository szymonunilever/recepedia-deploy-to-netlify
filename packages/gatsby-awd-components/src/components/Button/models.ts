import { ReactNode } from 'react';
import { Icon } from '../../models';

export interface ButtonCallback {
  // @ts-ignore
  (...arg): void;
}
export enum ButtonViewType {
  classic,
  icon,
}

export interface ButtonProps {
  className?: string;
  Icon?: Icon;
  IconSelected?: Icon;
  content?: AppContent.CTAContent;
  role?: string;
  isDisabled?: boolean;
  isSelected?: boolean;
  onClick?: ButtonCallback;
  onKeyDown?: (e: React.KeyboardEvent<HTMLElement>) => void;
  children?: ReactNode | ReactNode[];
  viewType?: ButtonViewType;
  hidden?: boolean;
  isToggle?: boolean;
  type?: 'button' | 'submit' | 'reset';
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  attributes?: Record<string, any>;
}
