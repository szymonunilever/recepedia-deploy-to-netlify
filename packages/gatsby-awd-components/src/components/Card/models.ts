import { ReactChild, ReactElement } from 'react';
import { Icon, RatingAndReviewsProvider, UnileverLibraryComponent } from '../../models';
import { ButtonProps } from '../Button';
import { RatingProps } from '../Rating';

export interface CardProps {
  key: string;
  content: CardContentProps;
  idPropertyName: string;
  ratingWidget?: ReactElement<RatingProps>
  imageSizes: string;
  className?: string;
  brand?: ReactChild;
  children?: ReactElement<ButtonProps> | ReactElement<ButtonProps>[];
}

export interface CardContentProps {
  fields:{
    slug: string
  };
  name?: string;
  title?: string;
  localImage?: Internal.LocalImage;
  images?: Internal.LocalImage[];
  averageRating?: number;
  [key: string]: any;
}

interface CardWrapper {
  key: string;
  children: ReactElement<CardProps>;
  ratingProvider: RatingAndReviewsProvider;
}

export interface RecipeCardWrapperProps extends CardWrapper{}

export interface ProductCardWrapperProps extends CardWrapper{
  mainImageIndex?: number;
}
