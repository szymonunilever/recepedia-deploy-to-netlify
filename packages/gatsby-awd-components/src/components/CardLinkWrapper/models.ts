import { ReactElement } from 'react';
import { CardProps, ProductCardWrapperProps, RecipeCardWrapperProps } from '../Card';
import { RecipeCardProps } from '../RecipeCard';

export interface CardLinkWrapperProps {
  key: string;
  isExternal?: boolean;
  title: string;
  slug: string;
  target?: string;
  href?: string;
  rel?: string;
  to?: string;
  'aria-label'?: string;
  className?: string;
  children: ReactElement<RecipeCardProps|CardProps|RecipeCardWrapperProps|ProductCardWrapperProps>;
}
