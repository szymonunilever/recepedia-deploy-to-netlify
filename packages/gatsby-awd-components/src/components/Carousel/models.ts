import { ReactChild, ReactElement } from 'react';
import { Icon } from '../../models';
import { RecipeCardProps } from '../RecipeListing/partials/models';

export interface CreateElement {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  (itemData: any): ReactElement<RecipeCardProps> | undefined;
}

export interface BreakpointProps {
  width: number;
  switchElementsBelowBreakpoint: number;
  switchElementsAfterBreakpoint: number;
  visibleElementsBelowBreakpoint: number;
  visibleElementsAboveBreakpoint: number;
}

export interface CarouselConfig {
  breakpoints?: BreakpointProps[];
  arrowIcon?: Icon;
}

export interface CarouselProps {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  list: any;
  createElementFunction: CreateElement;
  config: CarouselConfig;
  onVisibleElementsChanged?: Function;
}
