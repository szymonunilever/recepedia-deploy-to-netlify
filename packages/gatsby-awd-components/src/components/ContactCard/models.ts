import {
  titleLevel,
  UnileverLibraryComponent,
} from '../../models/globalModels';

export interface ContactCardProps extends UnileverLibraryComponent<AppContent.ContactCard> {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  Icon?: any;
  titleLevel: titleLevel;
}
