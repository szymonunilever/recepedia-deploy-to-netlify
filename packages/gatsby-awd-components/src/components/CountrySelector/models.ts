export interface LanguageEntry {
  value: string;
  label: string;
  path: string;
}

export interface CountrySelectorProps {
  selected?: LanguageEntry;
  list: LanguageEntry[];
  className?: string;
  flagSize?: FlagIconSize;
}

export type FlagIconSize = 'lg' | '2x' | '3x' | '4x' | '5x';
