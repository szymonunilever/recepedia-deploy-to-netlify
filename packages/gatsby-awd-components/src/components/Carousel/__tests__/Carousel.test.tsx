import { mount, ReactWrapper } from 'enzyme';
import React from 'react';
import { Carousel } from '../index';
import toJson from 'enzyme-to-json';
import { CarouselProps } from '../models';

describe('<Carousel />', () => {
  let wrapper: ReactWrapper;
  const props: CarouselProps = {
    list: [1, 2, 3],
    // eslint-disable-next-line react/display-name
    createElementFunction: () => <div>123</div>,
    config: {},
  };

  beforeEach(() => {
    wrapper = mount(<Carousel {...props} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('matches the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
