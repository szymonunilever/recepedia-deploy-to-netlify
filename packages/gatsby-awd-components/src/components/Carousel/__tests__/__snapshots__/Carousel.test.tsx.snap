// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Carousel /> matches the snapshot 1`] = `
<Carousel
  config={Object {}}
  createElementFunction={[Function]}
  list={
    Array [
      1,
      2,
      3,
    ]
  }
>
  <ProgressBar
    percentage={100}
  >
    <div
      className="progressBar progress-bar"
    >
      <progress
        aria-label="progress"
        className="progressBar__content progress-bar__content"
        max="100"
        role="progressbar"
        style={
          Object {
            "width": "100%",
          }
        }
        value="100"
      >
        100
        % complete
      </progress>
    </div>
  </ProgressBar>
  <div
    className="carousel carousel"
  >
    <div
      className="carousel__images carousel__images"
    >
      <div
        className="carousel__imagesTracker carousel__images-tracker"
        style={
          Object {
            "transform": "translateX(0%)",
            "width": "100%",
          }
        }
      >
        <div
          aria-hidden={false}
          className="carousel__item carousel__item"
          key="0"
          style={
            Object {
              "flexBasis": "33.333333333333336%",
              "transition": "all .5s",
              "visibility": "visible",
            }
          }
        >
          <div>
            123
          </div>
        </div>
        <div
          aria-hidden={false}
          className="carousel__item carousel__item"
          key="1"
          style={
            Object {
              "flexBasis": "33.333333333333336%",
              "transition": "all .5s",
              "visibility": "visible",
            }
          }
        >
          <div>
            123
          </div>
        </div>
        <div
          aria-hidden={false}
          className="carousel__item carousel__item"
          key="2"
          style={
            Object {
              "flexBasis": "33.333333333333336%",
              "transition": "all .5s",
              "visibility": "visible",
            }
          }
        >
          <div>
            123
          </div>
        </div>
      </div>
    </div>
  </div>
</Carousel>
`;
